pkg/scheduler/plugins/nodegroup/nodegroup.go

```diff
@@ -34,17 +34,21 @@ const (
 	PluginName       = "nodegroup"
 	NodeGroupNameKey = "volcano.sh/nodegroup-name"
 
-	BaseScore = 100
+	BaseScore        = 100
+	NonStrictPenalty = -2
 )
 
 type nodeGroupPlugin struct {
 	// Arguments given for the plugin
 	pluginArguments framework.Arguments
+	nonstrict       bool
```

Comment:
I prefer to set a strict flag instead of nonstricet, which is more understandable, and it's defaut value is true.